pkg/proxy/proxy.go

```diff
@@ -327,3 +321,28 @@ func (p *Proxy) authenticateRequest(req *http.Request) (*authenticator.Response,
 		User: u,
 	}, true, nil
 }
+
+func replaceTokenCookie(token string, req *http.Request) error {
+	tokenCookie, err := req.Cookie(ObotAccessTokenCookie)
+	if err != nil {
+		return fmt.Errorf("failed to get token cookie: %w", err)
+	}
+
+	tokenCookie.Value = token
+
+	otherCookies := make([]http.Cookie, 0, len(req.Cookies()))
+	for _, c := range req.Cookies() {
+		if c.Name != ObotAccessTokenCookie {
+			otherCookies = append(otherCookies, *c)
+		}
+	}
+
+	req.Header.Del("Cookie")
+
+	for _, c := range otherCookies {
+		req.AddCookie(&c)
+	}
```

Comment:
Could you do something like this so you only have to loop once?
```suggestion
	cookies := req.Cookie()
	req.Header.Del("Cookie")
	for _, c := range cookies {
		if c.Name != ObotAccessTokenCookie {
			req.AddCookie(c)
		}
	}
```